cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -563,6 +566,91 @@ private boolean needsReconfiguring(Reconciliation reconciliation, String connect
         }
     }
 
+    private Future<Map<String, Object>> maybeRestartConnector(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, CustomResource resource, Map<String, Object> status) {
+        if (hasRestartAnnotation(resource, connectorName)) {
+            log.debug("{}: Restarting connector {}", reconciliation, connectorName);
+            return apiClient.restart(host, port, connectorName)
+                    .compose(ignored -> removeRestartAnnotation(reconciliation, resource),
+                        throwable -> {
+                            // Ignore restart failures - just try again on the next reconcile
+                            log.warn("{}: Failed to restart connector {}. {}", reconciliation, connectorName, throwable.getMessage());
+                            return Future.succeededFuture();
+                        })
+                    .compose(ignored -> apiClient.statusWithBackOff(new BackOff(200L, 2, 10), host, port,
```

Comment:
Not sure I understand why do we call this at the end. Could you please explain?

Reply:
I'm guessing because ultimately `maybeCreateOrUpdateConnector` needs to return the status. But it would probably be better to make `pauseResume`, `maybeRestartConnector` and `maybeRestartConnectorTask` all return `Future<Void>` and make the call the `apiClient.statusWithBackOff` directly from `maybeCreateOrUpdateConnector`.

Reply:
Yeah, that is why I wondered about this. If it is about the status for the KafkaConnector status field, it should be probably called only at the end.

Reply:
Thanks for the review - Tom is correct here & I have updated to move the `apicClient.statusWithBackOff` call to the end of the compose chain(s).